com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/classification.compute

```diff
@@ -0,0 +1,136 @@
+        // It's more expensive... Need to profile and test for false positives in practice.
+        b = !IntervalsOverlap(tileEntityIndexRange, zBinEntityIndexRange);
+    }
+
+    return b;
+}
+
+/* ------------------------------ Implementation ---------------------------- */
+
+#ifndef PLATFORM_SUPPORTS_WAVE_INTRINSICS
+    groupshared uint gs_TileFeatureFlags;
+#endif
+
+[numthreads(TILE_SIZE * TILE_SIZE, 1, 1)]
+void main(uint3 threadID : SV_GroupThreadID, uint3 groupID : SV_GroupID)
+{
+    const uint  t         = threadID.x;
+    const uint2 tileCoord = groupID.xy;
+    const uint  eye       = groupID.z;
+
+    const uint  tile              = IndexFromCoordinate(tileCoord, TILE_BUFFER_DIMS.x);
+    const uint2 localPixelCoord   = DecodeMorton2D(t & (TILE_SIZE * TILE_SIZE - 1)); // Use AND to inform the compiler
+    const uint2 globalPixelCoord  = tileCoord * TILE_SIZE + localPixelCoord;
+    const uint2 clampedPixelCoord = min(globalPixelCoord, uint2(_ScreenSize.xy) - 1);
+
+#ifndef PLATFORM_SUPPORTS_WAVE_INTRINSICS
+    if (t == 0) // Avoid bank conflicts
+    {
+        gs_TileFeatureFlags = 0;
+    }
+#endif // PLATFORM_SUPPORTS_WAVE_INTRINSICS
+
+    uint featureFlags = 0;
+
+    // Unlit object, sky/background and forward opaque tag don't tag the StencilUsage.RequiresDeferredLighting bit
+    uint stencilVal = GetStencilValue(LOAD_TEXTURE2D_X(_StencilTexture, clampedPixelCoord));
+
+    // TODO: use coarse stencil here.
+    if ((stencilVal & STENCILUSAGE_REQUIRES_DEFERRED_LIGHTING) != 0)
+    {
+        featureFlags = g_BaseFeatureFlags; // Contain all lightFeatures or 0 (depends if we enable light classification or not)
+
+    #ifdef LIGHT_CLASSIFICATION
+        // TODO: optimize using LinearEyeDepth(float2 positionNDC, float deviceDepth, float4 invProjParam).
+        float          depth    = LoadCameraDepth(clampedPixelCoord);
+        PositionInputs posInput = GetPositionInput(clampedPixelCoord, _ScreenSize.zw, depth, UNITY_MATRIX_I_VP, UNITY_MATRIX_V);
+        uint           zBin     = ComputeZBinIndex(posInput.linearDepth);
+
+        if (!IsBinEmpty(tile, zBin, BOUNDEDENTITYCATEGORY_PUNCTUAL_LIGHT, eye))
+        {
+            featureFlags |= LIGHTFEATUREFLAGS_PUNCTUAL;
+        }
+
+        if (!IsBinEmpty(tile, zBin, BOUNDEDENTITYCATEGORY_AREA_LIGHT, eye))
+        {
+            featureFlags |= LIGHTFEATUREFLAGS_AREA;
+        }
+
+        if (!IsBinEmpty(tile, zBin, BOUNDEDENTITYCATEGORY_REFLECTION_PROBE, eye))
+        {
+            featureFlags |= LIGHTFEATUREFLAGS_ENV;
+        }
+    #endif // LIGHT_CLASSIFICATION
+
+    #ifdef MATERIAL_CLASSIFICATION
+        featureFlags |= MATERIAL_FEATURE_FLAGS_FROM_GBUFFER(clampedPixelCoord);
+    #endif // MATERIAL_CLASSIFICATION
+    }
+
+#if (defined(PLATFORM_SUPPORTS_WAVE_INTRINSICS) && (PLATFORM_LANE_COUNT >= (TILE_SIZE * TILE_SIZE)))
+    featureFlags = WaveActiveBitOr(featureFlags);
```

Comment:
isn't this only working if lane count == TILE_SIZE * TILE_SIZE  ?  (and not >= ) ?

Reply:
If it's greater then the remaining lanes should be inactive and thus not contribute to the result.